Submodule Sources/CLevelDB contains modified content
diff --git a/Sources/CLevelDB/include/leveldb/cache.h b/Sources/CLevelDB/include/leveldb/cache.h
index a94c683..bbbfc10 100644
--- a/Sources/CLevelDB/include/leveldb/cache.h
+++ b/Sources/CLevelDB/include/leveldb/cache.h
@@ -18,6 +18,8 @@
 #ifndef STORAGE_LEVELDB_INCLUDE_CACHE_H_
 #define STORAGE_LEVELDB_INCLUDE_CACHE_H_
 
+#ifdef __cplusplus
+
 #include <cstdint>
 
 #include "leveldb/export.h"
@@ -100,4 +102,6 @@ class LEVELDB_EXPORT Cache {
 
 }  // namespace leveldb
 
+#endif
+
 #endif  // STORAGE_LEVELDB_INCLUDE_CACHE_H_
diff --git a/Sources/CLevelDB/include/leveldb/comparator.h b/Sources/CLevelDB/include/leveldb/comparator.h
index a85b51e..21b36fb 100644
--- a/Sources/CLevelDB/include/leveldb/comparator.h
+++ b/Sources/CLevelDB/include/leveldb/comparator.h
@@ -5,6 +5,8 @@
 #ifndef STORAGE_LEVELDB_INCLUDE_COMPARATOR_H_
 #define STORAGE_LEVELDB_INCLUDE_COMPARATOR_H_
 
+#ifdef __cplusplus
+
 #include <string>
 
 #include "leveldb/export.h"
@@ -61,4 +63,6 @@ LEVELDB_EXPORT const Comparator* BytewiseComparator();
 
 }  // namespace leveldb
 
+#endif
+
 #endif  // STORAGE_LEVELDB_INCLUDE_COMPARATOR_H_
diff --git a/Sources/CLevelDB/include/leveldb/db.h b/Sources/CLevelDB/include/leveldb/db.h
index a13d147..69b14a6 100644
--- a/Sources/CLevelDB/include/leveldb/db.h
+++ b/Sources/CLevelDB/include/leveldb/db.h
@@ -5,6 +5,8 @@
 #ifndef STORAGE_LEVELDB_INCLUDE_DB_H_
 #define STORAGE_LEVELDB_INCLUDE_DB_H_
 
+#ifdef __cplusplus
+
 #include <cstdint>
 #include <cstdio>
 
@@ -164,4 +166,6 @@ LEVELDB_EXPORT Status RepairDB(const std::string& dbname,
 
 }  // namespace leveldb
 
+#endif
+
 #endif  // STORAGE_LEVELDB_INCLUDE_DB_H_
diff --git a/Sources/CLevelDB/include/leveldb/dumpfile.h b/Sources/CLevelDB/include/leveldb/dumpfile.h
index a58bc6b..0664444 100644
--- a/Sources/CLevelDB/include/leveldb/dumpfile.h
+++ b/Sources/CLevelDB/include/leveldb/dumpfile.h
@@ -5,6 +5,8 @@
 #ifndef STORAGE_LEVELDB_INCLUDE_DUMPFILE_H_
 #define STORAGE_LEVELDB_INCLUDE_DUMPFILE_H_
 
+#ifdef __cplusplus
+
 #include <string>
 
 #include "leveldb/env.h"
@@ -25,4 +27,6 @@ LEVELDB_EXPORT Status DumpFile(Env* env, const std::string& fname,
 
 }  // namespace leveldb
 
+#endif
+
 #endif  // STORAGE_LEVELDB_INCLUDE_DUMPFILE_H_
diff --git a/Sources/CLevelDB/include/leveldb/env.h b/Sources/CLevelDB/include/leveldb/env.h
index e00895a..b48bc19 100644
--- a/Sources/CLevelDB/include/leveldb/env.h
+++ b/Sources/CLevelDB/include/leveldb/env.h
@@ -13,6 +13,8 @@
 #ifndef STORAGE_LEVELDB_INCLUDE_ENV_H_
 #define STORAGE_LEVELDB_INCLUDE_ENV_H_
 
+#ifdef __cplusplus
+
 #include <cstdarg>
 #include <cstdint>
 #include <string>
@@ -414,4 +416,6 @@ class LEVELDB_EXPORT EnvWrapper : public Env {
 #endif  // defined(UNICODE)
 #endif  // defined(_WIN32) && defined(LEVELDB_DELETEFILE_UNDEFINED)
 
+#endif
+
 #endif  // STORAGE_LEVELDB_INCLUDE_ENV_H_
diff --git a/Sources/CLevelDB/include/leveldb/filter_policy.h b/Sources/CLevelDB/include/leveldb/filter_policy.h
index 49c8eda..1213a75 100644
--- a/Sources/CLevelDB/include/leveldb/filter_policy.h
+++ b/Sources/CLevelDB/include/leveldb/filter_policy.h
@@ -16,6 +16,8 @@
 #ifndef STORAGE_LEVELDB_INCLUDE_FILTER_POLICY_H_
 #define STORAGE_LEVELDB_INCLUDE_FILTER_POLICY_H_
 
+#ifdef __cplusplus
+
 #include <string>
 
 #include "leveldb/export.h"
@@ -69,4 +71,6 @@ LEVELDB_EXPORT const FilterPolicy* NewBloomFilterPolicy(int bits_per_key);
 
 }  // namespace leveldb
 
+#endif
+
 #endif  // STORAGE_LEVELDB_INCLUDE_FILTER_POLICY_H_
diff --git a/Sources/CLevelDB/include/leveldb/iterator.h b/Sources/CLevelDB/include/leveldb/iterator.h
index bb9a5df..52d97b2 100644
--- a/Sources/CLevelDB/include/leveldb/iterator.h
+++ b/Sources/CLevelDB/include/leveldb/iterator.h
@@ -15,6 +15,8 @@
 #ifndef STORAGE_LEVELDB_INCLUDE_ITERATOR_H_
 #define STORAGE_LEVELDB_INCLUDE_ITERATOR_H_
 
+#ifdef __cplusplus
+
 #include "leveldb/export.h"
 #include "leveldb/slice.h"
 #include "leveldb/status.h"
@@ -109,4 +111,6 @@ LEVELDB_EXPORT Iterator* NewErrorIterator(const Status& status);
 
 }  // namespace leveldb
 
+#endif
+
 #endif  // STORAGE_LEVELDB_INCLUDE_ITERATOR_H_
diff --git a/Sources/CLevelDB/include/leveldb/options.h b/Sources/CLevelDB/include/leveldb/options.h
index d755f46..9c53dd4 100644
--- a/Sources/CLevelDB/include/leveldb/options.h
+++ b/Sources/CLevelDB/include/leveldb/options.h
@@ -5,6 +5,8 @@
 #ifndef STORAGE_LEVELDB_INCLUDE_OPTIONS_H_
 #define STORAGE_LEVELDB_INCLUDE_OPTIONS_H_
 
+#ifdef __cplusplus
+
 #include <cstddef>
 
 #include "leveldb/export.h"
@@ -187,4 +189,6 @@ struct LEVELDB_EXPORT WriteOptions {
 
 }  // namespace leveldb
 
+#endif
+
 #endif  // STORAGE_LEVELDB_INCLUDE_OPTIONS_H_
diff --git a/Sources/CLevelDB/include/leveldb/slice.h b/Sources/CLevelDB/include/leveldb/slice.h
index 37cb821..08f4a07 100644
--- a/Sources/CLevelDB/include/leveldb/slice.h
+++ b/Sources/CLevelDB/include/leveldb/slice.h
@@ -15,6 +15,8 @@
 #ifndef STORAGE_LEVELDB_INCLUDE_SLICE_H_
 #define STORAGE_LEVELDB_INCLUDE_SLICE_H_
 
+#ifdef __cplusplus
+
 #include <cassert>
 #include <cstddef>
 #include <cstring>
@@ -111,4 +113,6 @@ inline int Slice::compare(const Slice& b) const {
 
 }  // namespace leveldb
 
+#endif
+
 #endif  // STORAGE_LEVELDB_INCLUDE_SLICE_H_
diff --git a/Sources/CLevelDB/include/leveldb/status.h b/Sources/CLevelDB/include/leveldb/status.h
index e327314..b0a57c5 100644
--- a/Sources/CLevelDB/include/leveldb/status.h
+++ b/Sources/CLevelDB/include/leveldb/status.h
@@ -13,6 +13,8 @@
 #ifndef STORAGE_LEVELDB_INCLUDE_STATUS_H_
 #define STORAGE_LEVELDB_INCLUDE_STATUS_H_
 
+#ifdef __cplusplus
+
 #include <algorithm>
 #include <string>
 
@@ -119,4 +121,6 @@ inline Status& Status::operator=(Status&& rhs) noexcept {
 
 }  // namespace leveldb
 
+#endif
+
 #endif  // STORAGE_LEVELDB_INCLUDE_STATUS_H_
diff --git a/Sources/CLevelDB/include/leveldb/table.h b/Sources/CLevelDB/include/leveldb/table.h
index a30e903..7e855f4 100644
--- a/Sources/CLevelDB/include/leveldb/table.h
+++ b/Sources/CLevelDB/include/leveldb/table.h
@@ -5,6 +5,8 @@
 #ifndef STORAGE_LEVELDB_INCLUDE_TABLE_H_
 #define STORAGE_LEVELDB_INCLUDE_TABLE_H_
 
+#ifdef __cplusplus
+
 #include <cstdint>
 
 #include "leveldb/export.h"
@@ -81,4 +83,6 @@ class LEVELDB_EXPORT Table {
 
 }  // namespace leveldb
 
+#endif
+
 #endif  // STORAGE_LEVELDB_INCLUDE_TABLE_H_
diff --git a/Sources/CLevelDB/include/leveldb/table_builder.h b/Sources/CLevelDB/include/leveldb/table_builder.h
index 85710c3..e000e33 100644
--- a/Sources/CLevelDB/include/leveldb/table_builder.h
+++ b/Sources/CLevelDB/include/leveldb/table_builder.h
@@ -13,6 +13,8 @@
 #ifndef STORAGE_LEVELDB_INCLUDE_TABLE_BUILDER_H_
 #define STORAGE_LEVELDB_INCLUDE_TABLE_BUILDER_H_
 
+#ifdef __cplusplus
+
 #include <cstdint>
 
 #include "leveldb/export.h"
@@ -90,4 +92,6 @@ class LEVELDB_EXPORT TableBuilder {
 
 }  // namespace leveldb
 
+#endif
+
 #endif  // STORAGE_LEVELDB_INCLUDE_TABLE_BUILDER_H_
diff --git a/Sources/CLevelDB/include/leveldb/write_batch.h b/Sources/CLevelDB/include/leveldb/write_batch.h
index 94d4115..78d11da 100644
--- a/Sources/CLevelDB/include/leveldb/write_batch.h
+++ b/Sources/CLevelDB/include/leveldb/write_batch.h
@@ -21,6 +21,8 @@
 #ifndef STORAGE_LEVELDB_INCLUDE_WRITE_BATCH_H_
 #define STORAGE_LEVELDB_INCLUDE_WRITE_BATCH_H_
 
+#ifdef __cplusplus
+
 #include <string>
 
 #include "leveldb/export.h"
@@ -80,4 +82,6 @@ class LEVELDB_EXPORT WriteBatch {
 
 }  // namespace leveldb
 
+#endif
+
 #endif  // STORAGE_LEVELDB_INCLUDE_WRITE_BATCH_H_
